File: src/main/kotlin/io/titandata/titan/providers/local/RemoteList.kt
/*
 * Copyright (c) 2019 by Delphix. All rights reserved.
 */

package io.titandata.titan.providers.local

import io.titandata.client.apis.RemotesApi

class RemoteList (
    private val remotesApi: RemotesApi = RemotesApi()
) {
    private val n = System.lineSeparator()
    
    fun list(container: String) {
        val remotes = remotesApi.listRemotes(container)
        System.out.printf("%-20s  %s${n}", "REMOTE", "PROVIDER")
        for (remote in remotes) {
            System.out.printf("%-20s  %s${n}", remote.name, remote.provider)
        }
    }
}

File: internal/app/providers/common/RemoteList.go
package common

import (
	"fmt"
	"github.com/titan-data/remote-sdk-go/remote"
	"strconv"
)

func RemoteList(repo string, port int)  {
	cfg.BasePath = "http://localhost:" + strconv.Itoa(port)

	remotes, _, _ := remotesApi.ListRemotes(ctx, repo)
	fmt.Printf("%-20s %-20s\n", "REMOTE", "URI") //TODO get proper os line separator
	for _, r := range remotes {
		url, _, _ := remote.Get(r.Provider).ToURL(r.Properties)
		fmt.Printf("%-20s %-20s\n", r.Name, url)
	}
}